templates/topic.html

```diff
@@ -4,18 +4,25 @@
 
 {{ define "body" }}
 
-<h2>{{ .Topic.Title }}</h2>
+<h2>
+{{ .Topic.Title }}
+    {{ if .Topic.IsClosed}}
+        <span style="color: dimgrey;font-weight:bold; font-size: large;" >[Closed]</span>
+    {{end}}
+</h2>
 
 <p>
     <a href="{{ .BasePath }}">Home</a> / <a href="{{ .BasePath }}categories/{{ .Category.CategoryID }}">{{ .Category.Name }}</a> / {{ .Topic.Title }}
 </p>
 
-<p style="text-align: right;">
-    <a class="btn-link" href="{{ .BasePath }}categories/{{ .Category.CategoryID }}/topics/{{ .Topic.TopicID }}/comments/new">
-        <svg xmlns="http://www.w3.org/2000/svg" x="0px" y="0px" width="12" height="12" viewBox="0 0 172 172" style=" fill:#000000;"><g fill="none" fill-rule="nonzero" stroke="none" stroke-width="1" stroke-linecap="butt" stroke-linejoin="miter" stroke-miterlimit="10" stroke-dasharray="" stroke-dashoffset="0" font-family="none" font-weight="none" font-size="none" text-anchor="none" style="mix-blend-mode: normal"><path d="M0,172v-172h172v172z" fill="none"></path><g fill="#ffffff"><path d="M85.83203,17.04323c-6.32845,0.09274 -11.38527,5.2949 -11.2987,11.62344v45.86667h-45.86667c-4.13529,-0.05848 -7.98173,2.11417 -10.06645,5.68601c-2.08471,3.57184 -2.08471,7.98948 0,11.56132c2.08471,3.57184 5.93115,5.74449 10.06645,5.68601h45.86667v45.86667c-0.05848,4.13529 2.11417,7.98173 5.68601,10.06645c3.57184,2.08471 7.98948,2.08471 11.56132,0c3.57184,-2.08471 5.74449,-5.93115 5.68601,-10.06645v-45.86667h45.86667c4.13529,0.05848 7.98173,-2.11417 10.06645,-5.68601c2.08471,-3.57184 2.08471,-7.98948 0,-11.56132c-2.08471,-3.57184 -5.93115,-5.74449 -10.06645,-5.68601h-45.86667v-45.86667c0.04237,-3.09747 -1.17017,-6.08033 -3.36168,-8.26973c-2.1915,-2.18939 -5.17553,-3.39907 -8.27296,-3.35371z"></path></g></g></svg>
-        New comment
-    </a>
-</p>
+{{ if eq .Topic.IsClosed false}}
```

Comment:
It's good to hide the button, but you also need to check in the GET and POST requests too and block there as well with a Forbidden or something.